lib/wasi/src/runtime/term.rs

```diff
@@ -0,0 +1,120 @@
+#![allow(unused_imports)]
+#[cfg(unix)]
+use {
+    libc::{
+        c_int, tcsetattr, termios, ECHO, ECHOE, ECHONL, ICANON, ICRNL, IEXTEN, ISIG, IXON, OPOST,
+        TCSANOW,
+    },
+    std::mem,
+    std::os::unix::io::AsRawFd,
+};
+
+#[cfg(unix)]
+pub fn io_result(ret: libc::c_int) -> std::io::Result<()> {
+    match ret {
+        0 => Ok(()),
+        _ => Err(std::io::Error::last_os_error()),
+    }
+}
+
+#[cfg(unix)]
+pub fn set_mode_no_echo() -> std::fs::File {
```

Comment:
This function is only used when `#[cfg(feature = "host-termios")]`. So perhaps should we place all the term into `#[cfg(feature = "host-termios")]` mode and raise a compiler error if not in unix?

Reply:
As commented above:

A)  this stuff does not belong on the Runtime trait at all
B) Even if it did, the config flags make no sense. This is a trait, so the code should just live in environment  specific implementations (like a Unix impl)

I bet this is the result of refactoring a struct into a trait...

Reply:
Refactored the code so these functions live in a submodule gated by the dependency.